packages/back-end/src/controllers/stripe.ts

```diff
@@ -4,106 +4,141 @@ import {
+    }
+
+    const qtyInDb = org.members.length + (org.invites?.length || 0);
+
+    if (restart) {
+      if (qty !== qtyInDb) {
+        throw new Error("Error with quantity of users");
+      }
+    } else {
+      if (qty !== qtyInDb + 1) {
+        throw new Error("Error with quantity of users");
       }
-      req.organization = await createOrganization(
-        req.email || "",
-        req.userId || "",
-        name,
-        ""
-      );
     }
 
-    // Create customer in Stripe if not exists
-    if (!req.organization.stripeCustomerId) {
-      const resp = await stripe.customers.create({
-        email: req.email || "",
-        name: req.name || "",
+    let stripeCustomerId: string;
+
+    if (org.stripeCustomerId) {
+      stripeCustomerId = org.stripeCustomerId;
+    } else {
+      const { id } = await stripe.customers.create({
         metadata: {
-          user: req.userId || "",
-          organization: req.organization.id,
+          growthBookId: org.id,
         },
       });
-      req.organization.stripeCustomerId = resp.id;
-
-      await updateOrganization(req.organization.id, {
-        stripeCustomerId: resp.id,
-      });
+      stripeCustomerId = id;
     }
 
-    // Start subscription trial without payment method
-    const subscription = await stripe.subscriptions.create({
-      customer: req.organization.stripeCustomerId,
-      coupon: STRIPE_DEFAULT_COUPON,
-      collection_method: "charge_automatically",
-      trial_from_plan: true,
-      metadata: {
-        user: req.userId || "",
-        organization: req.organization.id,
-      },
-      items: [
+    await updateOrganization(org.id, {
+      stripeCustomerId: stripeCustomerId,
+    });
+
+    const session = await stripe.checkout.sessions.create({
+      mode: "subscription",
+      payment_method_types: ["card"],
+      customer: stripeCustomerId,
+      line_items: [
         {
-          price: STRIPE_PRICE,
+          price: org?.priceId || STRIPE_PRICE,
           quantity: qty,
         },
       ],
+      allow_promotion_codes: true,
+      success_url: `${APP_ORIGIN}/settings/team`,
+      cancel_url: `${APP_ORIGIN}/settings/team`,
+    });
+    res.status(200).json({
+      status: 200,
+      session,
+    });
+  } catch (e) {
+    res.status(400).json({
+      status: 400,
+      message: e.message,
     });
+  }
+}
 
-    // Save in Mongo
-    await updateSubscription(subscription);
+export async function getSubscriptionData(req: AuthRequest, res: Response) {
+  try {
+    req.checkPermissions("organizationSettings");
 
-    res.status(200).json({ status: 200 });
+    const { org } = getOrgFromReq(req);
+
+    const subscriptionId = org.subscription?.id;
+
+    if (!subscriptionId) {
+      return res.status(200).json({
+        status: 200,
+        subscription: null,
+      });
+    }
+
+    const subscriptionData = await stripe.subscriptions.retrieve(
+      subscriptionId,
+      {
+        expand: ["plan"],
+      }
+    );
```

Comment:
Similar to my comment above, what's stopping us from just returning `org.subscription` without doing an api call to Stripe?  We're currently just storing a few of the subscription fields in the organization record (qty, trial dates, status, etc.).  We can add more there if needed.

Reply:
Old habits, I suppose - I've always received push back when suggesting we add fields to the DB. I've updated the organization modal to support these subscription fields so we don't have to hit Stripe's API and can instead just return `org.subscription`.